File: checkout/src/js/http/SuggestionApi.ts
import {HttpService} from "../common/services/HttpService";
import {CommonError} from "../model/common/CommonError";
import {SuggestionRequest} from "../model/v2/suggestion/SuggestionRequest";
import {SuggestionResponse} from "../model/v2/suggestion/SuggestionResponse";

export class SuggestionApi {
  constructor(private http: HttpService) {
  }

  getSuggestion(suggestionRequest: SuggestionRequest): Promise<SuggestionResponse> {
    return this.http.POST(`/v1/suggestion`, suggestionRequest);
  }
}

File: checkout-api/src/main/groovy/ish/oncourse/willow/service/SuggestionApi.java
package ish.oncourse.willow.service;

import ish.oncourse.willow.model.v2.suggestion.SuggestionRequest;
import ish.oncourse.willow.model.v2.suggestion.SuggestionResponse;

import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;

@Path("/")
public interface SuggestionApi  {

    @POST
    @Path("/v1/suggestion")
    @Produces({ "application/json" })
    @CollegeInfo
    SuggestionResponse getSuggestion(SuggestionRequest suggestionRequest);
}

